File: 4313-password-strength/password-strength.cpp
class Solution {
public:
    int passwordStrength(string password) 
    {
        unordered_set<char> st;
        int ans = 0;
        
        for(auto i : password)
        {
            if ( st.find(i) == st.end() && islower(i)) ans++;
            else if (st.find(i) == st.end() && isupper(i)) ans += 2;
            else if (st.find(i) == st.end() && isdigit(i)) ans += 3;
            else if (st.find(i) == st.end()) ans += 5;
            st.insert(i);
        }

        return ans;
        
    }
};

File: 4313-password-strength/password-strength.py
class Solution:
    def passwordStrength(self, password: str) -> int:
        st = 0
        sety = set()
        for i in password:
            if i not in sety and i.islower():
                st += 1
            elif i not in sety and i.isupper():
                st += 2
            elif i not in sety and i.isdigit():
                st += 3
            elif i not in sety:
                st += 5

            sety.add(i)
        
        return st


        